numpyro/distributions/util.py

```diff
@@ -361,23 +244,27 @@ def body_fn(val):
     return k - 1
 
 
+def poisson(key, rate, shape):
+    return _poisson(key, rate, shape)
+
+
 def _scatter_add_one(operand, indices, updates):
     return lax.scatter_add(operand, indices, updates,
                            lax.ScatterDimensionNumbers(update_window_dims=(),
                                                        inserted_window_dims=(0,),
                                                        scatter_dims_to_operand_dims=(0,)))
 
 
-@partial(jit, static_argnums=(1, 3))
-def multinomial_rvs(key, n, p, shape=()):
+@partial(jit, static_argnums=(2, 3))
+def _multinomial(key, p, n, shape=()):
```

Comment:
the pattern
```
@partial(jit, static_argnums=(2, 3))
def _f(a,b,c,d):
    ...

def f(a,b,c,d):
    return _f(a,b,c,d)
```
is used in `jax.random` to support different ways to set `args`, `kwargs` to samplers. Without this pattern, calling `f(a,b,c=c,d=d)` will trigger an error that `c` and `d` must be args, not kwargs.

Reply:
Hmm..what's the reason for this? Is this related to the change that kwargs are not treated as static args by default in jit?

Reply:
Yes, kwargs are not allowed in the positions of static args in the newer version (you can see that commit [here](https://github.com/google/jax/pull/595)).